src/Parsing/ParserState.php

```diff
@@ -345,6 +345,10 @@ public function consumeUntil(
         $start = $this->currentPosition;
 
         while (!$this->isEnd()) {
+            $comment = $this->consumeComment();
```

Comment:
We should cover this with a test.

Reply:
Yeah, it's a bit mysterious why the bug this is fixing suddenly reared it's head with this change.  Will need investigating as a separate issue.

Reply:
So how should we proceed here? Fix that bug first and then continue here? Or merge this PR first and than fix that bug (and cover it with tests)?

I tend to prefer the first, but am not completely sure.

WDYT?